src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala

```diff
@@ -88,7 +88,7 @@ class CoreModuleImpl @Inject() (
     eventStream,
     hostPort,
     crashStrategy,
-    richCuratorFramework.client,
+    richCuratorFramework.client.usingNamespace(null), // using non-namespaced client for leader-election
```

Comment:
hmm, can we go the other way? Start with no namespace, and then add it for the storage parts?

Reply:
I guess we could but should we? `/marathon/leader-curator` is the only exception - everything else *should* got into`/marathon/state`

Reply:
Personally, I'd prefer it, but I don't have an exact logical explanation as to why. I just find it a little surprising to see us set some namespace on the original connection, only to undo it later on for a different use case. There'd be less "overriding" or more "refining" if we started with a more general connection config, first.

Reply:
> set some namespace on the original connection, only to undo it later on for a different use case.

I'd say there is more potential to screw things up, by name-spacing (or rather forgetting) *every other* usage of the client *except* for leader-election. Again, a *namespace is the rule, no namespace is the exception*. But if you absolutely insist on it...